src/java/org/apache/cassandra/schema/CompressionParams.java

```diff
@@ -97,128 +109,185 @@
     // TODO: deprecated, should now be carefully removed. Doesn't affect schema code as it isn't included in equals() and hashCode()
     private volatile double crcCheckChance = 1.0;
 
-    public static CompressionParams fromMap(Map<String, String> opts)
-    {
-        Map<String, String> options = copyOptions(opts);
-
-        String sstableCompressionClass;
-
-        if (!opts.isEmpty() && isEnabled(opts) && !containsSstableCompressionClass(opts))
-            throw new ConfigurationException(format("Missing sub-option '%s' for the 'compression' option.", CLASS));
 
-        if (!removeEnabled(options))
-        {
-            sstableCompressionClass = null;
+    public enum CompressorType
+    {
+        lz4("LZ4Compressor"),
```

Comment:
@Claudenw what if names of these classes are changed of we make mistake typing them out? Would not it be better if we use `LZ4Compressor.class.getName()` instead?